lib/components/adapter.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @private
+ * @class Adapter
+ * @description Making requests to Shopify API.
+ * @requires request
+ * @requires bluebird
+ */
+'use strict';
+const request = require('request');
+const bluebird = require('bluebird');
```

Comment:
If I'm not mistaken `bluebird` is only used to promisify `request`, so we can probably use [`got`](https://github.com/sindresorhus/got) instead which has built in support for promises and is less bloated.